### ISSUE TITLE
ReqlDriveError

### ISSUE THREAD

**Issue by Sofiaj311** [grahamjenson/hapiger#2]
Hello, I'm trying your recommendation system HapiGER. I'm using RethinkDB. I'm tracking actions, persons and thinks from my website and sending them to HapiGER. That's working perfect, I can see all the data correctly stored in RethinkDB but when I ask HapiGER for a recommendation for a person with some history on the system (or none, it's the same effect), HapiGER doesn't work because it encounter an error with ReqlDriver and it can't connect to RethinkDB in order to do the query.
This is the console extract from my server (Ubuntu 12.04.05 LTS):

$ hapiger -v --es rethinkdb --esoptions '{"host":"devserver","port":28015,"db":"test"}' -n 10 -v
{ esm: 'rethinkdb',
  esmoptions: { host: 'devserver', port: 28015, db: 'test' },
  port: 3456,
  minimum_history_limit: 1,
  similar_people_limit: 25,
  related_things_limit: 10,
  recommendations_limit: '10',
  recent_event_days: 14,
  previous_actions_filter: [],
  compact_database_person_action_limit: 1500,
  compact_database_thing_action_limit: 1500,
  person_history_limit: 500,
  crowd_weight: 0 }
Creating a pool connected to devserver:28015
Starting Server on 3456
Unhandled Error { [ReqlDriverError: Unrecognized option `useOutdated` in `run`. Available options are readMode <string>, durability <string>, noreply <bool>, timeFormat <string>, groupFormat: <string>, profile <bool>, binaryFormat <string>, cursor <bool>, stream <bool>.]
  message: 'Unrecognized option `useOutdated` in `run`. Available options are readMode <string>, durability <string>, noreply <bool>, timeFormat <string>, groupFormat: <string>, profile <bool>, binaryFormat <string>, cursor <bool>, stream <bool>.' } ReqlDriverError: Unrecognized option `useOutdated` in `run`. Available options are readMode <string>, durability <string>, noreply <bool>, timeFormat <string>, groupFormat: <string>, profile <bool>, binaryFormat <string>, cursor <bool>, stream <bool>.
  at [object Object]._onTimeout (/usr/lib/node_modules/hapiger/node_modules/ger/node_modules/rethinkdbdash/lib/term.js:186:44)
  at Timer.listOnTimeout [as ontimeout](timers.js:121:15)

150928/061010.767, [response], http://devserver:3456: get /default/recommendations {"person":"4de204f11fe378b3","action":"view"} 500 (47ms)

I think it could be an error for incorrect compatibility between Driver and RethinkDB versions. I'm really interested in your work, GER system it's amazing and powerful, great work Graham.

Best regards, Sofia.

**Comment by grahamjenson** [comment]
Hey, I am not too familiar with the RethinkDB implementation (an issue I intend to fix soon) , as @thelinuxlich implemented that part of GER. Also, the latest version of GER does not support RethinkDB yet (another issue I intend to fix), as some quick changes have caused incompatibilities.

Both RethinkDB and the driver GER is using have been massively improved in the recent months and it is probably an incompatibility between old drivers new RethinkDB.

I will look into upgrading GER's RethinkDB driver, and HapiGER to use the new GER interface soon. If in the meantime if you could give me details on the version of RethinkDB you are using, I will try ensure that everything is compatible :)

**Comment by Sofiaj311** [comment]
Thanks for the quick answer!
I'm using RethinkDB 2.1.4, I tried with Postgre but I couldn't connect correctly because HapiGER show me an esm_url error, I tried several options for connection that I found at knex documentation. For information, it's the latest version of Postgre.

**Comment by grahamjenson** [comment]
I am upgrading GER to work with the latest RethinkDB version now, this might take a week or more. Once this is complete and tested, then HapiGER can be upgraded. Note: the next upgrade will will break the API, but overall the new version is much easier to work with. After this is completed, hopefully your issue will be resolved :)

**Comment by Sofiaj311** [comment]
There is no problem with the change of API, I will recode the specific part of the callings. I'm excited with test the new version. Thanks.

**Comment by grahamjenson** [comment]
Cool. I have upgraded GER to the new rethinkdb, it is a bit slower than before as @thelinuxlich has not had a look at it. I still have to upgrade HapiGER and the documentation.

I still think getting the postgres stuff working is probably the best idea, but the API will change either way, and after this get a bit more stable.

Cheers
Graham

**Comment by grahamjenson** [comment]
@Sofiaj311 So I have updated HapiGER to the new API 0.12  https://www.npmjs.com/package/hapiger. I have yet to update the website yet, that will come soon.

Although, the rethinkdb engine has been updated it is a bit slower now. I would recommend trying to get the Postgres engine working, it is the most tested.

If you have any other problems please raise an issue again :)
